### PR TITLE
JWT rule improvement

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add strong filter for JSON web token rule. It detects only JWT with header and open payloads (base64 encoded json). ecrypted JWT will not be detected
- Common registered words for JSON keys are used
- TODO: update and merge after https://github.com/Samsung/CredData/pull/157

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v4
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: jwt
```
Comment (babenek) [inline .github/workflows/benchmark.yml:26]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -66,7 +67,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:71]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -171,7 +173,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:177]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -348,7 +351,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:355]:
```suggestion
          repository: Samsung/CredData
```

**Review by babenek — COMMENTED** [review]
TODO: rollback after https://github.com/Samsung/CredData/pull/122

**Review by kmnls — DISMISSED** [review]
Agree, seems ok

File: credsweeper/rules/config.yaml [hunk 5]
```diff
@@ -304,16 +304,15 @@
 
 - name: JSON Web Token
   severity: medium
-  confidence: moderate
+  confidence: strong
```
Comment (xDizzix) [inline credsweeper/rules/config.yaml:307]: I think it a controversial statement that the JSON encoded in base64 is a security threat

File: credsweeper/filters/value_jwt_check.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+from credsweeper.utils import Util
+
+
+class ValueJwtCheck(Filter):
+    """JWT token check - simple"""
+
+    def __init__(self, config: Config = None) -> None:
+        pass
+
+    def run(self, line_data: LineData, target: AnalysisTarget) -> bool:
```
Comment (xDizzix) [inline credsweeper/filters/value_jwt_check.py:19]: The checker is very loose. It must be more strict, at least check the inner structure for the presence of some security-related keywords.

File: experiment/src/entropy_test.py [hunk 7]
```diff
@@ -100,5 +97,32 @@ def evaluate_avg(_args: Tuple[int, float, float]) -> Tuple[float, float]:
 # 31 = (4.5121865964712535, 0.1393228408491736)
 # 32 = (4.545556887485041, 0.13347416608982715)
 # 33 = (4.576938427997454, 0.1300362152603773)
+
+# 33 = (4.57601357276539, 0.13672208599146715)
```
Comment (xDizzix) [inline experiment/src/entropy_test.py:101]: Please describe what is this data for

**Review by xDizzix — CHANGES_REQUESTED** [review]
The rule is too loose

**Review by Yullia — APPROVED** [review]
Approved

**Review by Yullia — APPROVED** [review]

**Review by xDizzix — APPROVED** [review]

**Pull request merged**